### ISSUE TITLE
Add flip/mirror option to GridMap

### ISSUE THREAD

**Issue by matt-testing** [godotengine/godot-proposals#5883]
### Describe the project you are working on

3d sidecroller (arguably 2.5d) using gridmap to build levels composed of  3d tiles, used analogously to a 2d tilemap. 

### Describe the problem or limitation you are having in your project

To create something as simple as a tileable 2d box, a mesh library of 9 unique models is necessary. For more complex tilesets (i.e. including interior corners, holes, slopes) the number of models quickly becomes hard to parse in the gridmap tile selection panel, making 'drawing' complex shapes unintuitive. Having the option to mirror while 'drawing' the gridmap drastically reduces the number of necessary unique models to achieve similar results to 2d tilemaps. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

![sensical](https://user-images.githubusercontent.com/30683355/205002052-71ad6c09-5c58-469e-a7da-7ac3e7989392.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add mirroring options to the gridmap UI in conjunction with the current rotation options. 
Implementation was previously prototyped by @Varicus and @Xenoamor back in 2019, code examples, prototypes, and discussion are in the link below.

https://github.com/godotengine/godot/issues/31340#issuecomment-524817768

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It would probably be used with more frequency than the current 'rotate' options, since rotated tiles will show seams in ways that mirrored tiles will not.

### Is there a reason why this should be core and not an add-on in the asset library?

It should be a core function of Gridmap alongside tile rotation

**Comment by Mickeon** [comment]
I was about to say "Uhm, actually, this is a duplicate of https://github.com/godotengine/godot-proposals/issues/5158" but this is **GridMap**. I feel like this feature would suit GridMap even more than TileMap.

**Comment by matt-testing** [comment]
@Calinou can you clarify why you feel that is an "issue?" I can't think of a use case for the ability to mirror a 3d model without also mirroring its UV information, the end result being that UV mapping is still correctly aligned with the model.

**Comment by Calinou** [comment]
> @Calinou can you clarify why you feel that is an "issue?" I can't think of a use case for the ability to mirror a 3d model without also mirroring its UV information, the end result being that UV mapping is still correctly aligned with the model.

If you have a tile with a cobblestone texture for instance, you may not want the cobblestone texture to appear flipped on flipped tiles.

**Comment by matt-testing** [comment]
It seems like World Triplanar Mapping would be the solution for models whose texture map is desired to display independent of mesh rotation/scale. I would expect mirrored meshlib objects with Triplanar mapping to not mirror the texture, in this proposal's implementation.

**Comment by Megalomaniak** [comment]
> I would expect mirrored meshlib objects with Triplanar mapping to not mirror the texture, in this proposal's implementation.

Indeed.

Not directly related to this but I'd also like to be able to have child nodes for/per tiles when creating GridMaps. Could be useful for having Position 3D nodes as spawn points for procedural generation and/or pseudo randomly picked and placed objects. Could also then include such a child of the tile for a floor mesh that could be placed at the location of that Position3D node without taking the rest of it's transform into account.

In fairness though there's already [a kind of a way](https://www.youtube.com/watch?v=h1K2k0Bw4CA) of achieving all of that.

**Comment by matt-testing** [comment]
This proposal is more about enhancing the current functionality of gridmaps, however there is a larger discussion to be had about how gridmap could be used as a more general-purpose level design tool. You might be interested in the discussion her: https://github.com/godotengine/godot-proposals/issues/3518#issue-1047741752
